connect/runtime/src/main/java/org/apache/kafka/connect/util/LoggingContext.java

```diff
@@ -181,11 +187,16 @@ public static LoggingContext forOffsets(ConnectorTaskId id) {
      * @param connectorName the name of the connector; may not be null
      * @param scope the scope; may not be null
      * @param taskNumber the 0-based task number; may be null if there is no associated task
+     * @param prefix optional prefix for the value; may be null
      * @return the prefix; never null
      */
-    static String prefixFor(String connectorName, Scope scope, Integer taskNumber) {
+    static String prefixFor(String connectorName, Scope scope, Integer taskNumber, String prefix) {
         StringBuilder sb = new StringBuilder();
         sb.append("[");
+        if (prefix != null) {
+            sb.append(prefix);
+            sb.append("|");
+        }
```

Comment:
I've proposed changes to the logging context before and they've been met with (valid) concerns about causing issues with logging analysis tools that are set up to parse contexts by, e.g., regex.

I hate to say it, but I think this kind of change requires a KIP since it entails risk of breaking users' setups.

Reply:
Thank you, I was trying to minimize the impact of this (only affects MM2 dedicated mode), but I understand the concern, will kick off a KIP